python/tk_multi_publish2/api/item.py

```diff
@@ -0,0 +1,952 @@
+            published item in Shotgun and is meant to represent it visually. The
+            ``thumbnail`` should not be confused with the item's ``icon`` which
+            is for use only in the publisher UI and represents the type of item
+            being published.
+        """
+
+        # nothing to do if running without a UI
+        if not sgtk.platform.current_engine().has_ui:
+            return None
+
+        # defer import until needed and to avoid issues when running without UI
+        from sgtk.platform.qt import QtGui
+
+        if self._thumbnail_path and not self._thumbnail_pixmap:
+            # we have a path but haven't yet created the pixmap. create it
+            try:
+                self._thumbnail_pixmap = QtGui.QPixmap(self._thumbnail_path)
+            except Exception, e:
+                logger.warning(
+                    "%r: Could not load icon '%s': %s" %
+                    (self, self._icon_path, e)
+                )
+
+        if self._thumbnail_pixmap:
+            return self._thumbnail_pixmap
+        elif self.parent:
+            return self.parent.thumbnail
+        else:
+            return None
+
+    @thumbnail.setter
+    def thumbnail(self, pixmap):
+        """Sets the thumbnail """
+        self._thumbnail_pixmap = pixmap
+
+    @property
+    def thumbnail_enabled(self):
+        """
+        Boolean property to indicate whether thumbnails can be interacted with
+        for items displayed in a publish UI.
+
+        * If ``True``, thumbnails will be visible and editable in the publish UI
+          (via screen capture).
+        * If ``False`` and a thumbnail has been set via the :meth:`thumbnail`
+          property, the thumbnail will be visible but screen capture will be
+          disabled.
+        * If ``False`` and no thumbnail has been specified, no thumbnail will
+          appear in the UI.
+        """
+        return self._thumbnail_enabled
+
+    @thumbnail_enabled.setter
+    def thumbnail_enabled(self, enabled):
+        # setter for thumbnail_enabled
+        self._thumbnail_enabled = enabled
+
+    @property
+    def thumbnail_explicit(self):
+        """
+        Boolean property to indicate that thumbnail has been explicitly set.
+        When this flag is on, the summary thumbnail should be ignored for this
+        this specific item.
+        """
+        return self._thumbnail_explicit
+
+    @thumbnail_explicit.setter
+    def thumbnail_explicit(self, enabled):
+        """Setter for _thumbnail_explicit."""
+        self._thumbnail_explicit = enabled
+
+    @property
+    def type_spec(self):
+        """
+        The type specification for this item. This specification follows a
+        hierarchical dot notation. For example, 'file', 'file.image',
+        'file.quicktime' or 'maya_scene'. This specification is up to the client
+        code to define in terms of formatting and meaning.
```

Comment:
we do have some opinion about this - and we are setting a base line of type specs via the configs. I think it could be worth including our existing ones as a starting point?

Reply:
I've updated the description to note this and provide a more detailed overview with examples of how the default integration uses this property.